From ef568ac3630138a9fb1c633f65db7e68c049c1e7 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Wed, 21 Jan 2015 11:12:25 +0200
Subject: [PATCH 0102/1240] smp: changed the SMP affinity setup to adjust to
 new Aurora model

	This entire SMP setup is temporary in u-boot and should
	be handled by the PSCI firmware. That's why its executed
	under enable_wa.

Change-Id: I2132a162d12ab5dffcdec96f2b1ff468c826235d
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/16124
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm/cpu/armv8/start.S | 58 +++++++++++-----------------------------------
 1 file changed, 13 insertions(+), 45 deletions(-)

diff --git a/arch/arm/cpu/armv8/start.S b/arch/arm/cpu/armv8/start.S
index 78947d2..d669943 100644
--- a/arch/arm/cpu/armv8/start.S
+++ b/arch/arm/cpu/armv8/start.S
@@ -21,7 +21,7 @@
 .globl	_start
 _start:
 #ifdef CONFIG_PALLADIUM
-	bl	fixup_affinity
+	bl	enable_wa
 #endif
 	b	reset
 
@@ -179,54 +179,22 @@ ENTRY(c_runtime_cpu_setup)
 ENDPROC(c_runtime_cpu_setup)
 
 #ifdef CONFIG_PALLADIUM
-ENTRY(fixup_affinity)
+ENTRY(enable_wa)
+	/* This setup is temporary, eventually it should go into
+	 * the PSCI firmware */
 
-/*
- * Configure snoop affinity.
- * in UP each master snoops only itself
- * For cpu 0 set DVM affiinity with SMMU & self
- */
-	ldr	x0, =0xf0004210
-	ldr	w1, =0x2030003
-	str	w1, [x0]
-
-	ldr     x0, =0xf0004214
-	ldr	w1, =0x2030003
-	str     w1, [x0]
-
-	ldr     x0, =0xf0004218
-	ldr     w1, =0x4
-	str     w1, [x0]
-
-	ldr     x0, =0xf000421c
-	ldr     w1, =0x8
-	str     w1, [x0]
-
-	ldr     x0, =0xf0004220
-	ldr     w1, =0x0
+	/* Activate Affinity between cpu0 and cpu1 */
+	ldr     x0, =0xf0004264
+	ldr     w1, =0x3
 	str     w1, [x0]
 
-	ldr     x0, =0xf0004224
-	ldr     w1, =0x0
-	str     w1, [x0]
-
-	ldr     x0, =0xf0004228
-	ldr     w1, =0x0
-	str     w1, [x0]
-
-	ldr     x0, =0xf000422c
-	ldr     w1, =0x0
-	str     w1, [x0]
-
-	ldr     x0, =0xf0004230
-	ldr     w1, =0x0
-	str     w1, [x0]
-
-	ldr     x0, =0xf0004234
-	ldr     w1, =0x0
-	str     w1, [x0]
+	/* Activate Affinity in CA-57 configuration
+	 * Enable the SMPEN bit in CPUECTLR_EL1 */
+	mrs x0, S3_1_c15_c2_1
+	orr x0, x0, #0x40
+	msr S3_1_c15_c2_1, x0
 
 	ret
 
-ENDPROC(fixup_affinity)
+ENDPROC(enable_wa)
 #endif /* CONFIG_PALLADIUM */
-- 
1.9.1

